flocker/node/agents/test/test_blockdevice.py

```diff
@@ -5001,3 +5004,40 @@ def test_get_device_path_until_detach(self):
 
         self.assertRaises(UnattachedVolume,
                           self.cache.get_device_path, attached_id1)
+
+
+def make_icloudapi_tests(
+        blockdevice_api_factory,
+):
+    """
+    :param blockdevice_api_factory: A factory which will be called
+        with the generated ``TestCase`` during the ``setUp`` for each
+        test and which should return a provider of both ``IBlockDeviceAPI``
+        and ``ICloudAPI`` to be tested.
+
+    :returns: A ``TestCase`` with tests that will be performed on the
+       supplied ``IBlockDeviceAPI``/``ICloudAPI`` provider.
+    """
+    class Tests(SynchronousTestCase):
```

Comment:
Shouldn't these be _asynchronous_ tests? They are running threads and rely on what looks like the real reactor.